application/models/Survey.php

```diff
@@ -2069,4 +2068,20 @@ public function setToInherit()
         }
     }
 
+    /**
+     * Find all public surveys
+     * @return Survey[]
+     */
+    public static function findAllPublic()
+    {
+        $surveys = self::model()->findAll();
```

Comment:
This won't scale. Filter in the query. :laughing:

Reply:
Also it doesn't seem to be called statically in the change above?

Reply:
> This won't scale. Filter in the query.

It is not possible to flter in the query as `public` could be an inherited value. That's the main reason of the bug. I could try to narrow the query a bit more still (Public and Inherited). Though, a loop for reviewing one by one the inherited ones will be needed

Reply:
Can it be both inactive and public?

Reply:
Looks like it filters by active? `Survey::model()->active()->open()->with('languagesettings')->findAllPublic()`

You should probably change the entire line, current way makes no sense.

Reply:
Not sure what `active()` actually calls here.

Reply:
I was trying to substitute the AR scope with a new method that goes through attributes.
Seems I missed something. Will review it.

Reply:
Thanks, ping me when you want a new review.